SmartDeviceLinkTests/DevAPISpecs/SDLVersionSpec.m

```diff
@@ -0,0 +1,103 @@
+#import <Quick/Quick.h>
+#import <Nimble/Nimble.h>
+
+#import "SDLSyncMsgVersion.h"
+#import "SDLVersion.h"
+
+QuickSpecBegin(SDLVersionSpec)
+
+describe(@"a version object", ^{
+    __block SDLVersion *testVersion = nil;
+    __block NSUInteger major = 5; __block NSUInteger minor = 4; __block NSUInteger patch = 0;
+
+    beforeEach(^{
+        testVersion = nil;
+    });
+
+    describe(@"initializers", ^{
+        context(@"created from numbers", ^{
+            it(@"should match the parameters", ^{
+                testVersion = [[SDLVersion alloc] initWithMajor:major minor:minor patch:patch];
+                expect(testVersion.major).to(equal(major));
+                expect(testVersion.minor).to(equal(minor));
+                expect(testVersion.patch).to(equal(patch));
+            });
+        });
+
+        context(@"created from a string", ^{
+            NSString *goodVersionString = [NSString stringWithFormat:@"%lu.%lu.%lu", major, minor, patch];
+            NSString *badVersionStringShort = @"1.";
+            NSString *badVersionStringLong = @"3.1.0.0";
+
+            it(@"should throw with a bad short string", ^{
+                expectAction(^{
+                    testVersion = [[SDLVersion alloc] initWithString:badVersionStringShort];
+                }).to(raiseException());
+            });
+
+            it(@"should throw with a bad long string", ^{
+                expectAction(^{
+                    testVersion = [[SDLVersion alloc] initWithString:badVersionStringLong];
+                }).to(raiseException());
+            });
+
+            it(@"should match the good string", ^{
+                testVersion = [[SDLVersion alloc] initWithString:goodVersionString];
+                expect(testVersion.major).to(equal(major));
+                expect(testVersion.minor).to(equal(minor));
+                expect(testVersion.patch).to(equal(patch));
+            });
+
+            fit(@"should fail with negative numbers", ^{
```

Comment:
1. Remove the "f" from "it"
2. The test cases stop because an assertion is thrown. You can prevent this by telling the test manager to expect an assertion.

```
expectAction(^{
    testVersion = [[SDLVersion alloc] initWithString:@"-1.-2.-3"];
}).to(raiseException());
```